geniza/annotations/views.py

```diff
@@ -33,6 +41,52 @@ def __init__(self, *args, **kwargs):
         super().__init__(content_type=self.content_type, *args, **kwargs)
 
 
+def parse_annotation_data(request):
+    """
+    For annotation create and update methods, parse json data from request in order
+    to set/update associated footnote. Returns content dict and footnote object.
+    """
+    json_data = json.loads(request.body)
+
+    # get manifest and source URIs and resolve to source and document
+    manifest_uri = json_data["target"]["source"]["partOf"]["id"]
+    source_uri = json_data["dc:source"]
+    source_id = Source.id_from_uri(source_uri)
+    document_id = document_id_from_manifest_uri(manifest_uri)
+    document_contenttype = ContentType.objects.get_for_model(Document)
```

Comment:
`KeyError` right now, but I suppose it would be better to handle this as a 400 bad request HTTP error—we would expect these from anything returned from Tahqiq.

Reply:
yes, I agree — better to return a 400 than a 500!